pkg/manager/impl/execution_manager.go

```diff
@@ -213,24 +215,42 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask) runti
 	return runtimeInterfaces.TaskResourceSet{CPU: cpuLimit, Memory: memoryLimit}
 }
 
+// This type stores the locations of where a CPU and memory resource are defined in a []core.Resources_ResourceEntry.
+type taskResourceIndices struct {
+	cpuIndex    int
+	memoryIndex int
+}
+
+func newTaskResourceIndices() taskResourceIndices {
+	return taskResourceIndices{
+		cpuIndex:    -1,
+		memoryIndex: -1,
+	}
+}
+
+func (i taskResourceIndices) isValid() bool {
+	return i.cpuIndex >= 0 && i.memoryIndex >= 0
+}
+
 func assignResourcesIfUnset(ctx context.Context, identifier *core.Identifier,
```

Comment:
this function makes me hate go.

Reply:
why

Reply:
haha any ugliness here is on me, not the language :P

Reply:
isn't this pretty common lol